integritybackend/actions.py

```diff
@@ -207,235 +207,286 @@ def create(self, asset_fullpath, jwt_payload, meta):
-            asset_helper.path_for(collection_id, "c2pa-custom", output=True),
-            asset_helper,
-        )
-
-    def _add(self, asset_fullpath, output_dir, asset_helper):
-        # Create temporary files to work with.
-        tmp_asset_file = asset_helper.get_tmp_file_fullpath(".jpg")
-        time.sleep(1)
-        _logger.info("File size: %s", os.path.getsize(asset_fullpath))
-        shutil.copy2(asset_fullpath, tmp_asset_file)
-
-        # Copy asset to both the internal and shared asset directories.
-        internal_asset_file = asset_helper.get_internal_file_fullpath(tmp_asset_file)
-        shutil.move(tmp_asset_file, internal_asset_file)
-        if output_dir is not None:
-            shutil.copy2(internal_asset_file, output_dir)
-        _logger.info("New asset file added: %s", internal_asset_file)
-
-        return internal_asset_file
+        # TODO: change function to take just org_id as param
+        action_name = "c2pa-proofmode"
+        org_id = org_config["id"]
+        asset_helper = AssetHelper(org_id)
+        file_util = FileUtil()
 
-    def _update(self, asset_fullpath, claim, output_dir, asset_helper):
-        # Create temporary files to work with.
-        tmp_asset_file = asset_helper.get_tmp_file_fullpath(".jpg")
-        tmp_claim_file = asset_helper.get_tmp_file_fullpath(".json")
+        # Get configs
+        collection = config.ORGANIZATION_CONFIG.get_collection(org_id, collection_id)
+        action = config.ORGANIZATION_CONFIG.get_action(org_id, collection_id, action_name)
+        action_params = action.get("params")
 
-        # Parse file name to get internal name for parent file.
-        file_name, file_extension = os.path.splitext(os.path.basename(asset_fullpath))
-        parent_file = os.path.join(
-            asset_helper.get_assets_internal(),
-            file_name.partition("_")[0].partition("-")[0] + file_extension,
-        )
+        # Get paths
+        action_dir = asset_helper.path_for_action(collection_id, action_name)
+        action_output_dir = asset_helper.path_for_action_output(collection_id, action_name)
+        action_tmp_dir = asset_helper.path_for_action_tmp(collection_id, action_name)
 
-        # Inject update claim and read back from file.
-        time.sleep(1)
-        _logger.info("File size: %s", os.path.getsize(asset_fullpath))
-        shutil.copy2(asset_fullpath, tmp_asset_file)
-        _logger.info("Searching for parent file: %s", parent_file)
-        if not os.path.isfile(parent_file):
-            raise Exception(f"Expected {parent_file} to be a file, but it isn't")
+        # Verify and copy zip
+        input_zip_sha = os.path.splitext(os.path.basename(zip_path))[0]
+        if input_zip_sha != file_util.digest_sha256(zip_path):
+            raise Exception(f"SHA-256 of ZIP does not match file name: {zip_path}")
+        tmp_zip = shutil.copy2(zip_path, action_tmp_dir)
+        bundle_name = f"{input_zip_sha}-images"
```

Comment:
I feel ZIP verification logic should be centralized to a place before actions occur, otherwise every action needs to re-verify. We probably want violating ZIPs to fail "completely" rather than succeeding for a subset of actions.